src/components/SubmissionsCarousel.tsx

```diff
@@ -125,29 +157,28 @@ export const SubmissionsCarousel: React.FC<{
           left: newIndex * slider.offsetWidth,
           behavior: behavior ?? "smooth",
         })
-        setCurrentIndex(newIndex)
-        setLastDisplayedSubmissionId(filteredSubmissions[newIndex].submissionId)
+        // setCurrentIndex(newIndex)
+        // setLastDisplayedSubmissionId(submissions[newIndex].submissionId)
       }
     },
-    [filteredSubmissions, slideCount],
+    [slideCount],
   )
-  useEffect(() => {
-    if (!lastDisplayedSubmissionId) return
-
-    const index = filteredSubmissions.findIndex(
-      (submission) => submission.submissionId === lastDisplayedSubmissionId,
-    )
-    if (index !== -1) {
-      goToSlide(index, "instant")
-    } else {
-      setLastDisplayedSubmissionId(filteredSubmissions[0]?.submissionId ?? null)
-    }
-  }, [lastDisplayedSubmissionId, filteredSubmissions, goToSlide])
+  //   useEffect(() => {
```

Comment:
This effect is essential for the SubmissionInspector to be properly usable without having to search for the last viewed submission each time we change a filter. If you want I can have a look at how I can properly integrate this with your newly added features:)